### PR TITLE
Made cross-browser architectural changes

### PR DESCRIPTION
**Pull request by rohandudam**
Made cross-browser architectural changes:
- Refactored cross-browsers code and created standalone/separate python file for each cross-browser 
- Made cross-browser integration and disintegration easy

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -45,9 +44,20 @@ def test_obj(base_url, browser, browser_version, os_version, os_name, remote_fla
             test_obj.set_rp_logger(reportportal_service)
 
         yield test_obj
+
         #Teardown
-        test_obj.wait(3)
-        test_obj.teardown()
+        def fin():
```
Comment (shivahari) [inline conftest.py:49]: Do we need to add this finalizer?
Can we not use a `finally` block in `try-catch`?

File: page_objects/core_helpers/selenium_action_objects.py [hunk 2]
```diff
@@ -290,9 +290,18 @@ def drag_and_drop(self, source_locator, target_locator):
...
 
-    def teardown(self):
+    def teardown(self, test_status = None):
         "Tears down the driver"
-        self.driver.quit()
-        self.reset()
-
-
+        import os
+        if os.getenv('REMOTE_BROWSER_PLATFORM') == 'LT' and test_status is not None:
```
Comment (shivahari) [inline page_objects/core_helpers/selenium_action_objects.py:296]: Do we need to modify the existing `teardown` method to use `execute_script` method?
Can we:
1. Add `self.driver` to `LambdaTestRunner` object and implement a method to update `lambda-status`?
2. Use `testobj.driver` in `conftest.py` to run `execute_script` method?

File: page_objects/drivers/local_browsers.py [hunk 3]
```diff
@@ -72,3 +72,31 @@ def headless_chrome():
         local_driver = webdriver.Chrome(options=options)
 
         return local_driver
+
+    def app_details(self, desired_capabilities, app_package, app_activity):
```
Comment (shivahari) [inline page_objects/drivers/local_browsers.py:76]: Can we have the `Mobile` driver setup methods in a separate object instead of `LocalBrowsers` ?

**Review by shivahari — CHANGES_REQUESTED** [review]
Please make the suggested fixes @rohandudam 
If you feel if any of the suggested change is redundant ♻️  or incorrect ❌ , please respond to that comment

File: conftest.py [hunk 4]
```diff
@@ -15,10 +15,9 @@
 load_dotenv()
 
 @pytest.fixture
-def test_obj(base_url, browser, browser_version, os_version, os_name, remote_flag, testrail_flag, tesults_flag, test_run_id, remote_project_name, remote_build_name, testname, reportportal_service, interactivemode_flag):
+def test_obj(request, base_url, browser, browser_version, os_version, os_name, remote_flag, testrail_flag, tesults_flag, test_run_id, remote_project_name, remote_build_name, testname, reportportal_service, interactivemode_flag):
```
[the remark this answers is not in the document]
Comment (rohandudam) [reply conftest.py:18]: yes

File: integrations/cross_browsers/lambdatest_runner.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        lambdatest_options = {}
+        lambdatest_options = self.lambdatest_credentials(lambdatest_options)
+        lambdatest_options["build"] = remote_build_name
+        lambdatest_options["project"] = remote_project_name
+        lambdatest_options["name"] = testname
+        lambdatest_options["video"] = True
+        lambdatest_options["visual"] = True
+        lambdatest_options["network"] = True
+        lambdatest_options["w3c"] = True
+        lambdatest_options["plugin"] = "python-pytest"
```
[the remark this answers is not in the document]
Comment (rohandudam) [reply integrations/cross_browsers/lambdatest_runner.py:38]: Sure

File: page_objects/core_helpers/selenium_action_objects.py (hunk 2 above)
Comment (rohandudam) [reply page_objects/core_helpers/selenium_action_objects.py]: Done.

File: page_objects/drivers/local_browsers.py (hunk 3 above)
Comment (rohandudam) [reply page_objects/drivers/local_browsers.py:76]: As per our discussion, renamed file name to local options

**Review by shivahari — APPROVED** [review]

**Pull request merged**